lineapy/plugins/pipeline_writers.py

```diff
@@ -0,0 +1,208 @@
+
+    def write_pipeline_files(self) -> None:
+        self._write_docker(
+            template_name="script_dockerfile.jinja",
+            template_params={
+                "pipeline_name": self.pipeline_name,
+                "python_version": get_system_python_version(),
+            },
+        )
+
+
+class AirflowPipelineWriter(BasePipelineWriter):
+    """
+    Class for pipeline file writer. Corresponds to "AIRFLOW" framework.
+    """
+
+    def __init__(
+        self, dag_config: Optional[AirflowDagConfig] = {}, *args, **kwargs
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.dag_config = dag_config or {}
+
+    def _write_dag(self) -> None:
+        dag_flavor = self.dag_config.get(
+            "dag_flavor", AirflowDagFlavor.PythonOperatorPerSession
+        )
+
+        if dag_flavor == AirflowDagFlavor.PythonOperatorPerSession:
+            DAG_TEMPLATE = load_plugin_template(
+                "airflow_dag_PythonOperatorPerSession.jinja"
+            )
+            session_functions = [
+                f"run_session_including_{session_artifacts._get_first_artifact_name()}"
+                for session_artifacts in self.session_artifacts_sorted
+            ]
+            dependencies = {
+                session_functions[i + 1]: {session_functions[i]}
+                for i in range(len(session_functions) - 1)
+            }
+            task_graph = TaskGraph(
+                nodes=session_functions,
+                mapping={f: f for f in session_functions},
+                edges=dependencies,
+            )
+            full_code = DAG_TEMPLATE.render(
+                DAG_NAME=self.pipeline_name,
+                MODULE_NAME=self.pipeline_name + "_module",
+                OWNER=self.dag_config.get("owner", "airflow"),
+                RETRIES=self.dag_config.get("retries", 2),
+                START_DATE=self.dag_config.get("start_date", "days_ago(1)"),
+                SCHEDULE_INTERVAL=self.dag_config.get(
+                    "schedule_interval", "*/15 * * * *"
+                ),
+                MAX_ACTIVE_RUNS=self.dag_config.get("max_active_runs", 1),
+                CATCHUP=self.dag_config.get("catchup", "False"),
+                tasks=session_functions,
+                task_dependencies=task_graph.get_airflow_dependency(),
+            )
+            full_code = prettify(full_code)
+
+            # Write out file
+            file = self.output_dir / f"{self.pipeline_name}_dag.py"
+            file.write_text(prettify(full_code))
+            logger.info("Generated DAG file %s", file)
+        else:
+            raise ValueError(
+                f'"{dag_flavor}" is an invalid airflow dag flavor.'
+            )
+
+    def write_pipeline_files(self) -> None:
+        self._write_dag()
+        self._write_docker(
+            template_name="dockerfile.jinja",
+            template_params={
+                "pipeline_name": self.pipeline_name,
```

Comment:
discussed with @yoonspark move to base class. use template method to add steps for daughter classes.

Reply:
Done